### PR TITLE
added DOMPurify from https://github.com/cure53/DOMPurify

### PR DESCRIPTION
**Pull request by davetayls**

### CODE REVIEW

**Comment by vvakame** [comment]
*dompurify/dompurify.d.ts*

check list

* [ ] is collect [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)?
  * https://www.npmjs.com/package/dompurify
  * http://bower.io/search/?q=dompurify
  * others?
* [X] has a [test file](http://definitelytyped.org/guides/contributing.html#tests)? (dompurify/dompurify-tests.ts or others)
* [x] pass the Travic-CI test?

File: dompurify/dompurify.d.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+// Project: https://github.com/cure53/DOMPurify
+// Definitions by: Dave Taylor <http://davetayls.me>
+// Definitions: https://github.com/borisyankov/DefinitelyTyped
+
+interface IDOMPurify {
+  sanitize(s:string):string;
+  addHook(hook:string, cb:(currentNode:Element, data:any, config:any) => Element):void;
+}
+
+declare var DOMPurify:IDOMPurify;
```
Comment (vvakame) [inline dompurify/dompurify.d.ts]: please move this line into `declare module`. see https://github.com/borisyankov/DefinitelyTyped/blob/master/gulp/gulp.d.ts#L281-L284

Comment (davetayls) [inline dompurify/dompurify.d.ts]: only issue with this is that when you include this in the browser with a script tag it creates a global `DOMPurify` namespace on the window object which is why I included this statement

Comment (CSharpFan) [inline dompurify/dompurify.d.ts]: @vvakame https://github.com/borisyankov/DefinitelyTyped/blob/master/lodash/lodash.d.ts#L6 Lodash does the same. 

It avoids having to do `(<any>window).xxx`.

Comment (vvakame) [inline dompurify/dompurify.d.ts]: okey, I understand.

**Comment by vvakame** [comment]
@davetayls thanks mate!

**Pull request merged**